File: android/2048/src/main/java/com/dreampany/play2048/service/NotifyService.java
package com.dreampany.play2048.service;

import androidx.annotation.NonNull;

import com.dreampany.play2048.vm.NotifyViewModel;
import com.dreampany.framework.api.service.BaseJobService;
import com.firebase.jobdispatcher.JobParameters;

import javax.inject.Inject;

/**
 * Created by Hawladar Roman on 7/22/2018.
 * Dreampany Ltd
 * dreampanymail@gmail.com
 */
public class NotifyService extends BaseJobService {

    @Inject
    NotifyViewModel vm;
    @Override
    protected boolean doJob(@NonNull JobParameters job) {
        vm.notifyIf();
        return true;
    }

    @Override
    protected boolean done(@NonNull JobParameters job) {
        vm.clear();
        return true;
    }
}

File: android/lca/src/main/kotlin/com/dreampany/lca/service/NotifyService.kt
package com.dreampany.lca.service

import com.dreampany.frame.api.service.BaseJobService
import com.dreampany.lca.vm.NotifyViewModel
import com.firebase.jobdispatcher.JobParameters
import javax.inject.Inject

/**
 * Created by Roman-372 on 7/30/2019
 * Copyright (c) 2019 bjit. All rights reserved.
 * hawladar.roman@bjitgroup.com
 * Last modified $file.lastModified
 */
class NotifyService : BaseJobService() {

    @Inject
    internal lateinit var vm: NotifyViewModel

    override fun doJob(job: JobParameters): Boolean {
        vm.notifyIf()
        return true
    }

    override fun done(job: JobParameters): Boolean {
        vm.clear()
        return true
    }
}